Graphics/GraphicsEngineWebGPU/include/FenceWebGPUImpl.hpp

```diff
@@ -56,7 +57,14 @@ class FenceWebGPUImpl final : public FenceBase<EngineWebGPUImplTraits>
     /// Implementation of IFence::Wait() in WebGPU backend.
     void DILIGENT_CALL_TYPE Wait(Uint64 Value) override final;
 
-    void SetCompletedValue(Uint64 Value);
+    void AppendSyncPoints(const std::vector<RefCntAutoPtr<ISyncPointWebGPU>>& SyncPoints, Uint64 EventOnComplete);
+
+    void RequestedValue(Uint64 Value);
+
+private:
+    using SyncPointEvent = std::pair<Uint64, std::vector<RefCntAutoPtr<ISyncPointWebGPU>>>;
+    std::atomic<Uint64>        m_RequestedFenceValue{0};
+    std::deque<SyncPointEvent> m_SyncPoints;
```

Comment:
There should be a list of sync points for each signalled value.
Something like `std::map<Uint64, std::vector<SyncPoint>>`
Map will keep them sorted, so it will be easy to iterate.

Reply:
Why use a map if the deque is already sorted?